File: app/src/main/java/com/adapter/demo/TitleItemVHFactory.java
package com.adapter.demo;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.adapter.BViewHolder;
import com.adapter.BViewHolderFactory;

/**
 * 添加人：  Tom Hawk
 * 添加时间：2018/7/24 9:45
 * 功能描述：
 * <p>
 * 修改人：  Tom Hawk
 * 修改时间：2018/7/24 9:45
 * 修改内容：
 */
public class TitleItemVHFactory extends BViewHolderFactory {

    @SuppressWarnings("unchecked")
    public BViewHolder createViewHolder(@NonNull LayoutInflater inflater, @Nullable ViewGroup parent, @NonNull Object item) {
        return new TitleItemVH(inflater, parent);
    }

    final class TitleItemVH extends BViewHolder<TitleItem> {
        private TextView tvText;

        TitleItemVH(@NonNull LayoutInflater inflater, @Nullable ViewGroup parent) {
            super(inflater, parent, false, R.layout.item_text);
            tvText = itemView.findViewById(R.id.tvText);
        }

        @Override
        public void setContents(@NonNull TitleItem item, boolean selectable) {
            tvText.setText(item.getTitle());
        }
    }
}

File: app/src/main/java/com/adapter/demo/TitleItemVHFactory.kt
package com.adapter.demo

import android.view.LayoutInflater
import android.view.ViewGroup
import com.adapter.BViewHolder
import com.adapter.BViewHolderFactory
import kotlinx.android.synthetic.main.item_text.view.*

/**
 * 添加人：  Tom Hawk
 * 添加时间：2018/7/24 9:45
 * 功能描述：
 *
 *
 * 修改人：  Tom Hawk
 * 修改时间：2018/7/24 9:45
 * 修改内容：
 */
class TitleItemVHFactory : BViewHolderFactory() {

    override fun createViewHolder(
            inflater: LayoutInflater,
            parent: ViewGroup?,
            item: Any
    ): BViewHolder<out Any> {
        return TitleItemVH(inflater, parent)
    }

    private inner class TitleItemVH(inflater: LayoutInflater, parent: ViewGroup?) :
            BViewHolder<TitleItem>(inflater, parent, R.layout.item_text) {

        override fun setContents(item: TitleItem, payloads: MutableList<Any>?) {
            itemView.tvText.text = item.title
        }
    }
}